render/id.go

```diff
@@ -14,6 +24,11 @@ func MakeEndpointID(hostID, addr, port string) string {
 	return makeID("endpoint", hostID, addr, port)
 }
 
+// MakePseudoEndpointID makes a pseudo endpoint node ID for rendered nodes.
+func MakePseudoEndpointID(hostID, addr, port string) string {
+	return makeID("pseudo-endpoint", hostID, addr, port)
+}
```

Comment:
Well the whole point is to have the pseudo nodes have endpoints as their children.  I guess there are two alternatives:
- introduce the pseudo nodes in the process mapping instead of the endpoint mapper
- make endpoint mapper add the 'report' endpoint nodes (from the selector) as children of the endpoints & pseudo nodes, and make the connection table code look for them instead

Can you think of any other alternatives?  Which do you prefer?